### ISSUE TITLE
Macro name conflict

### ISSUE THREAD

**Issue by zhuoqiang** [SergiusTheBest/plog#25]
While it is convenience to use the short name macro however they are very easy to conflict with other code, especially:
- `LOG`
- `LOGD`

It's better that plog provide a longer macro name with `PLOG_` prefix or something to avoid such conflict and let user to define there short hand name if needed:

``` c++
#include <plog/Log.h>

#define MY_LOG PLOG_LOG

MY_LOG(info) << "Will not conflict any more :)";
```

**Comment by SergiusTheBest** [comment]
Hi there!

It's too late to change the interface. And unfortunately C preprocessor has no ability to define a macro name with another macro (so I could make LOG macro prefix customizable). Thus the only option is to use the modified version of `Log.h` header or undefining conflicting macros.

By the way from what library are the conflicting `LOG` definitions?

**Comment by zhuoqiang** [comment]
Hi SergiusTheBest,

The conficting happened in our private project on android platform, which we had defined our own LOGD etc macros for logging.

I understood the public API should not be broken. There's anothor option:
Could there be a config MACRO so that when defined all the LOG macro will be prefixed, something like:

```c++

#ifdef PLOG_PREFIX_LOG_NAME
    #define PLOG_LOG ...
    ...

#else
    #define LOG ...
    ...

#endif
```
Or, to elimiate code duplication further:


```c++

// define all log with prefix anyway
#define PLOG_LOG ...
...

#ifndef PLOG_PREFIX_LOG_NAME_ONLY
    #define LOG PLOG_LOG
    ...
#endif
```
Of cause by default the `PLOG_PREFIX_LOG_NAME` wil not be defined. And user could define the MACRO if they want to use the prefix macro only.

**Comment by allannk** [comment]
I will second this, and mention that Google Tensorflow uses LOG as a macro name as well. I modified your Log.h to rename all macros PLOG<_...> instead, although it is not that beautiful a fix :)

**Comment by gordhub** [comment]
These names: LOG_INFO, LOG_WARNING, LOG_DEBUG
conflict with system headers <syslog.h> or <sys/syslog.h> depending on operating system.

It's pretty quick and easy to change all the macro names starting with LOG to start with PLOG instead (and that's what I've done here myself). However, I understand that the existing names are already out the door and just changing them will break code for existing users. 

It's a little more work, but very possible, to make the suppression of the existing LOG* names optional.

**Comment by SergiusTheBest** [comment]
@gordhub Oh, conflicting with syslog.h is not good.

**Comment by gordhub** [comment]
@SergiusTheBest agreed. Would you like me to submit  a PR (tomorrow, I'm headed for bed now)?

**Comment by SergiusTheBest** [comment]
@gordhub Yes, sure.

**Comment by SergiusTheBest** [comment]
Implemented in #129. It uses `PLOG_` instead of `PLOG_LOG_` that is shorter.
